File: Compat/src/main/java/me/ste/stevesseries/base/ReflectionUtil.java
package me.ste.stevesseries.base;

/**
 * @deprecated For backwards compatibility only.
 */
@Deprecated
public class ReflectionUtil {
    public static final String NMS_VERSION = me.ste.stevesseries.base.util.ReflectionUtil.INSTANCE.getNMS_VERSION();
    public static final String NMS_PACKAGE = me.ste.stevesseries.base.util.ReflectionUtil.INSTANCE.getNMS_PACKAGE();
    public static final String OBC_PACKAGE = me.ste.stevesseries.base.util.ReflectionUtil.INSTANCE.getOBC_PACKAGE();

    private ReflectionUtil() {}

    public static <T> Class<T> resolveClass(String classPackage, String clazz) throws ClassNotFoundException {
        return (Class<T>) me.ste.stevesseries.base.util.ReflectionUtil.INSTANCE.resolveClass(classPackage, clazz);
    }
}

File: Compat/src/main/kotlin/me/ste/stevesseries/base/util/ReflectionUtil.kt
package me.ste.stevesseries.base.util

import org.bukkit.Bukkit

@Deprecated("For backwards compatibility only.")
object ReflectionUtil {
    private val packageName = Bukkit.getServer().javaClass.getPackage().name

    val NMS_VERSION = packageName.substring(packageName.lastIndexOf('.') + 1)
    val NMS_PACKAGE = "net.minecraft.server.${NMS_VERSION}"
    val OBC_PACKAGE = "org.bukkit.craftbukkit.${NMS_VERSION}"

    fun resolveClass(javaPackage: String, className: String) = Class.forName("$javaPackage.$className")
}